src/actions/mod.rs

```diff
@@ -447,26 +447,38 @@ impl ActionHandler {
                 return;
             }
         };
-        let config = self.fmt_config.lock().unwrap();
-        let mut config = config.get_rustfmt_config().clone();
+
+        let range_whole_file = ls_util::range_from_vfs_file(&self.vfs, path);
+        let mut config = self.fmt_config.lock().unwrap().get_rustfmt_config().clone();
         if !config.was_set().hard_tabs() {
             config.set().hard_tabs(!opts.insert_spaces);
         }
         if !config.was_set().tab_spaces() {
             config.set().tab_spaces(opts.tab_size as usize);
         }
 
+        if let Some(r) = selection {
+            let range = ls_util::range_to_rls(r).one_indexed();
+            let file_lines = format!(r#"[{{
+                "file": "stdin",
+                "range": [{}, {}]
+            }}]"#, range.row_start.0, range.row_end.0);
```

Comment:
Could you use a struct and serialise it, rather than using format! here please? (Ideally, you could use a struct from Rustfmt, but it's ok to create your own here if that is not possible).